validator/src/main.rs

```diff
@@ -963,11 +964,13 @@ pub fn main() {
         .map(BlockstoreRecoveryMode::from);
 
     // Canonicalize ledger path to avoid issues with symlink creation
-    let _ = fs::create_dir_all(&ledger_path);
-    let ledger_path = fs::canonicalize(&ledger_path).unwrap_or_else(|err| {
-        eprintln!("Unable to access ledger path: {err:?}");
-        exit(1);
-    });
+    let ledger_path = create_and_canonicalize_directories(&[ledger_path])
+        .unwrap_or_else(|err| {
+            eprintln!("Unable to access ledger path: {err}");
```

Comment:
Thinking about making the logs easy to grep, it would be nice if the message contains `Error`. Since this is an `eprintln!()` instead of an `error!()`, what do you think about prepending `Error` to the existing for:
```
"Error: unable to access ledger path: {err}"
```